decidim-admin/app/commands/decidim/admin/update_resource_permissions.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Admin
+    # This command gets called when permissions for a resource not related with
+    # a component are updated in the admin panel.
+    class UpdateResourcePermissions < Rectify::Command
+      # Public: Initializes the command.
+      #
+      # form     - The form from which the data in this resource comes from.
+      # resource - The resource to update.
+      def initialize(form, resource)
+        @form = form
+        @resource = resource
+      end
+
+      # Public: Sets the permissions for a resource.
+      #
+      # Broadcasts :ok if created, :invalid otherwise.
+      def call
+        return broadcast(:invalid) unless form.valid?
+
+        transaction do
+          update_permissions
+        end
+
+        broadcast(:ok)
+      end
+
+      private
+
+      attr_reader :form, :resource
+
+      def configured_permissions
+        form.permissions.select do |_, permission|
+          selected_handlers(permission).any?
+        end
+      end
+
+      def update_permissions
+        permissions = configured_permissions.inject({}) do |result, (key, value)|
+          handlers_content = {}
+
+          selected_handlers(value).each do |handler_key|
```

Comment:
Maybe use also an inject here instead of `each` and defining the hash before?

Reply:
Done!